src/lando/main/management/commands/start_landing_worker.py

```diff
@@ -46,11 +46,23 @@ def _handle(self, worker: Worker, repo_type: str):
             # Check if any associated repos are unsupported, raise exception if so.
             repo.raise_for_unsupported_repo_scm(repo_type)
             scm = repo.get_scm()
-            scm.prepare_repo(repo.pull_path)
+            repos_to_handle = []
+            try:
+                scm.prepare_repo(repo.pull_path)
+                repos_to_handle.append(repo)
+            except Exception as e:
+                logger.warning(
+                    "Fail to prepare repo, skipping...",
+                    extra={
+                        "exception": e,
+                        "origin": repo.pull_path,
+                        "scm": scm.scm_type,
+                    },
+                )
 
         # Continue with starting the worker.
         try:
-            landing_worker = LandingWorker(worker.enabled_repos)
+            landing_worker = LandingWorker(repos_to_handle)
```

Comment:
If there's a temporary failure for whatever reason, this would mean that we have to force-restart the worker in order to pick up additional repos. I think it makes sense to consider that in the handling of these errors, for example, by detecting which repos failed and maybe retrying in the future.

Reply:
Yes.

Adding a retry would be substantially more work here, but a log message on temporary failures may not be sufficient in prod... I guess we could add a config option to determine whether we want to fail gracefully or not?